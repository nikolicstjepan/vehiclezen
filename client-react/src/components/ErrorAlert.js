import React from "react";

export default function ErrorAlert({ error }) {
  if (!error) {
    return null;
  }

  return (
    <div className="alert alert-danger m-3 position-fixed bottom-0 end-0" role="alert">
      {error}
    </div>
  );
}
