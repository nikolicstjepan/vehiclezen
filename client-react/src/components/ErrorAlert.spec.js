import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import ErrorAlert from "./ErrorAlert";

describe("ErrorAlert", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<ErrorAlert error="test" />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("returns null if no error", () => {
    const component = shallow(<ErrorAlert />);

    expect(component.html()).toBe(null);
  });

  test("returns alert with error text when error", () => {
    const component = shallow(<ErrorAlert error="test" />);

    expect(component.find(".alert").text()).toBe("test");
  });
});
