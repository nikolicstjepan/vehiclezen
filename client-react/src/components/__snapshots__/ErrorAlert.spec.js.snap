// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert snapshot renders 1`] = `
<div
  className="alert alert-danger m-3 position-fixed bottom-0 end-0"
  role="alert"
>
  test
</div>
`;
